### PR TITLE
Added export of `CPM_LAST_PACKAGE_NAME`

### PR DESCRIPTION
**Pull request by CraigHutchinson**
+ Updated docs respectively

#400

### CODE REVIEW

**Comment by CraigHutchinson** [comment]
image
@iboB Any idea on this one?

**Comment by CraigHutchinson** [comment]
Confirmed as working: 
image

File: README.md [hunk 1]
```diff
@@ -73,6 +73,7 @@ After calling `CPMAddPackage`, the following variables are defined in the local
 - `<dependency>_SOURCE_DIR` is the path to the source of the dependency.
 - `<dependency>_BINARY_DIR` is the path to the build directory of the dependency.
 - `<dependency>_ADDED` is set to `YES` if the dependency has not been added before, otherwise it is set to `NO`.
+- CPM_LAST_PACKAGE_NAME is set to `<dependency>`. For single-argument syntax this will be the automatically determined name.
```
Comment (TheLartians) [inline README.md:76]: Rephrased to be more clear imo and formatted CMake identifiers as code.

```suggestion
- `CPM_LAST_PACKAGE_NAME` is set to the determined name of the last added dependency (equivalent to `<dependency>`).
```

File: test/unit/local_dependency/ModuleCMakeLists.txt.in [hunk 2]
```diff
@@ -32,3 +32,4 @@ include(@CPM_PATH@/testing.cmake)
 ASSERT_TRUTHY(@TEST_DEPENDENCY_NAME@_ADDED)
 ASSERT_DEFINED(@TEST_DEPENDENCY_NAME@_SOURCE_DIR)
 ASSERT_DEFINED(@TEST_DEPENDENCY_NAME@_BINARY_DIR)
+ASSERT_EQUAL(CPM_LAST_PACKAGE_NAME "@TEST_DEPENDENCY_NAME@")
```
Comment (TheLartians) [inline test/unit/local_dependency/ModuleCMakeLists.txt.in:35]: I believe the variable should be expanded here and the other tests below.

```suggestion
ASSERT_EQUAL("${CPM_LAST_PACKAGE_NAME}" "@TEST_DEPENDENCY_NAME@")
```

**Review by TheLartians — COMMENTED** [review]
Thanks for creating the PR and also adding tests! 

Right now the test seems a bit unfocussed, however. I think a better location for checking the correct value of `CPM_LAST_PACKAGE_NAME` might be after each `CPMAddPackage` call in the [`dependency_properties.cmake`](https://github.com/cpm-cmake/CPM.cmake/blob/10bf25a8113e2003f259536bc88ad7bc81eff8ff/test/unit/dependency_properties.cmake#L11) unit test.

**Comment by TheLartians** [comment]
Note that you can run the unit tests locally using the following commands.

```bash
cmake -Stest -Bbuild/test
cmake --build build/test --target test-verbose
```

And also auto-format the code style style using [cmake-format](https://github.com/cheshirekow/cmake_format) as below. 

```bash
cmake -Stest/style -Bbuild/style
cmake --build build/style --target fix-format
```

File: test/unit/local_dependency/ModuleCMakeLists.txt.in (hunk 2 above)
Comment (CraigHutchinson) [reply test/unit/local_dependency/ModuleCMakeLists.txt.in:35]: Agreed, much better 💯

File: README.md (hunk 1 above)
Comment (CraigHutchinson) [reply README.md]: Agreed, much better 💯

File: test/unit/local_dependency/ModuleCMakeLists.txt.in (hunk 2 above)
Comment (CraigHutchinson) [reply test/unit/local_dependency/ModuleCMakeLists.txt.in:35]: Agree. Will look into these changes.

**Comment by CraigHutchinson** [comment]
> Note that you can run the unit tests locally using the following commands.

@TheLartians Thanks for this info, invaluable :)

File: test/unit/local_dependency/ModuleCMakeLists.txt.in (hunk 2 above)
Comment (CraigHutchinson) [reply test/unit/local_dependency/ModuleCMakeLists.txt.in]: Done

**Comment by CraigHutchinson** [comment]
I have updated and fixed the additional tests. However, hopefully as an aside, I get failure on this test prior & post my additional changes 
image

**Review by iboB — DISMISSED** [review]
I think it looks good

**Review by TheLartians — COMMENTED** [review]
Looks good, could you run `cmake-format` or the `fix-format` target to update the code style?

**Comment by TheLartians** [comment]
> EDIT2: It would also be useful if the UNit-test used a separate or temporary CPM_Cache location as it will currently be affected if/as there may already be an existing cache for packages of the name used in the test under some circumstances

Sounds like a good idea, though it may be even better to move all "unit-tests" that build a full project into the more extensive integration test suite.

**Comment by CraigHutchinson** [comment]
> Looks good, could you run `cmake-format` or the `fix-format` target to update the code style?

Not made much progress here yet, getting an error and not sure how to proceed 'yet': `RuntimeError: Failed to parse C:\development\CPM.cmake\.cmake-format as any of yaml, json, or python`

**Comment by iboB** [comment]
It seems that the style checks are still failing, though to be honest, I don't see why

**Comment by TheLartians** [comment]
Tried it on my machine and it removed a trailing whitespace. @CraigHutchinson did you install the required dependencies? That should be easy using pip:

```bash
pip3 install clang-format==14.0.6 cmake_format==0.6.11 pyyaml
```

**Review by TheLartians — APPROVED** [review]

**Review by iboB — APPROVED** [review]

**Comment by CraigHutchinson** [comment]
@TheLartians thanks for sorting the extra space, I shall try harder next time! As it stands I had issues with the fix-cmake in that it was changing line-endings in ALL the source files, I would gather in omitting this the space fix was also lost.
I haven't investigated yet, but my expectation is the issue should be git config related as the message is "LF will be replaced by CRLF the next time Git touches it"

**Pull request merged**